tests/grpc/utils/testService.ts

```diff
@@ -178,6 +179,46 @@ function createTestService({
         callback(null, message);
       }
     },
+    testStreamFail: async (
+      call: grpc.ServerWritableStream<utilsPB.EchoMessage, utilsPB.EchoMessage>,
+    ): Promise<void> => {
+      const genWritable = grpcUtils.generatorWritable(call);
+      try {
+        const echoMessage = new utilsPB.EchoMessage().setChallenge('Hello!');
+        for (let i = 0; i < 10; i++) {
+          await sleep(100);
+          await genWritable.write(echoMessage);
+        }
+        const message = call.request.getChallenge();
+        if (message === 'exit') process.exit(0);
+        if (message === 'kill') process.kill(process.pid);
+        if (message === 'sigkill') process.kill(process.pid, 9);
+        for (let i = 0; i < 10; i++) {
+          await sleep(100);
+          await genWritable.write(echoMessage);
+        }
+        await genWritable.write(null);
+      } catch (err) {
+        await genWritable.throw(err);
+      }
+    },
+    testUnaryFail: async (
+      call: grpc.ServerUnaryCall<utilsPB.EchoMessage, utilsPB.EchoMessage>,
+      callback: grpc.sendUnaryData<utilsPB.EchoMessage>,
+    ): Promise<void> => {
+      try {
+        // Wait a long time and then close the server.
+        await sleep(2000);
+        const message = call.request.getChallenge();
+        if (message === 'exit') process.exit(0);
+        if (message === 'kill') process.kill(process.pid);
+        if (message === 'sigkill') process.kill(process.pid, 9);
+        await sleep(2000);
+        callback(null, new utilsPB.EchoMessage().setChallenge('hello!'));
+      } catch (err) {
+        callback(grpcUtils.fromError(err));
+      }
+    },
```

Comment:
Let's call these `unaryFail` and `serverStreamFail` to align with the names of the other calls on this `testService`.

Also please add inline comments on each type of failure detailing what kind of failure it is.

Does this align with the different failure tests according to the diagrams in https://github.com/MatrixAI/js-polykey/issues/224#issuecomment-1033257102